python/tk_maya/menu_generation.py

```diff
@@ -294,6 +294,20 @@ def add_command_to_menu(self, menu):
             
         pm.menuItem(**params)
         
+    def __exception_trap_callback_wrapper(self, callback):
```

Comment:
Yepp totally - but i don't think it's about this wrapper - the whole menu system should be a framework. This has been on the todo list for an extended period of time and that means we could do lots of awesome stuff with menus and menu placement too - right now it's really crude.